cloudevents/extensions/verifiability.md

```diff
@@ -0,0 +1,290 @@
+verification material on the producer side and also perform the verification on
+the consumer side. This is the appropriate choice for any type of tool that is
+used by other entities, for example open source.
+
+**Note:** If an intermediary party modifies an event, they are considered the
+producer of a new event. They MUST create an updated verification material and
+the consumer(s) MUST recognize them as a trustworthy producer.
+
+## Verifiability Implementations
+
+While this proposal is agnostic to the specific implementations, we set the
+following rules for verifiability implementations:
+
+A verifiability implementation **MUST**:
+
+* transport the verification material in the same message as the event
+* use the two relevant context attributes in this proposal
+
+A verifiability implementation **SHOULD**:
+
+* avoid depending on canonicalization (and its
+[unnecessarily large attack surface](https://github.com/secure-systems-lab/dsse/blob/master/background.md#motivation))
+* transport the verification material detached from the event payload
+(this helps avoid canonicalization)
+
+A verifiability implementation **MUST FAIL** on:
+
+* Duplication of a relevant context attribute (e.g. two `verificationmaterial`
+  attributes being set in a message)
+* One relevant context attribute being present but the other missing (e.g.
+  `verificationmaterial` being set but `verificationmaterialtype` missing)
+* Unknown material type (e.g. the implementation only attempts to use the
+  verification material if it knows how to do so)
+* Invalid material (e.g. the verification material did not match the received
+  event)
+
+In order to get a verifiability implementation into the official CloudEvents
+SDK, the following requirements are set.
+
+## Verifiability Implementation Proposals
+
+A proposal for adding a verifiability implementation to the CE SDK **MUST:**
+
+* provide specs on how to implement verification
+* provide test vectors
+* provide a verification coverage table (see Verification coverage table)
+
+Test vectors ensure that implementations are consistent which does NOT affect verifiability. 
+
+A proposal for adding a verifiability implementation to the CE SDK **SHOULD:**
+
+* provide information about they handle attributes, and if relevant, normalization
+
+## Example
+
+To illustrate, let’s walk through how to implement a verification
+implementation CloudEvents provide test vectors as well as a verification
+coverage table.
+
+### Verification Implementation
+
+This is an example message containing a CloudEvent directly from the spec. It
+happens to be an HTTP structured mode message in JSON format. Our goal is to
+come up with the most insecure but also most concise verification
+implementation imaginable: one based on a reversed digest which we will call
+`rev`!  The `rev` implementation will only verify whether or not the
+payload, `data`, is valid and not additional attributes for the sake of
```

Comment:
...and not whether the other context attributes are valid, for the sake of simplicity.


yes?